src/Sulu/Component/HttpCache/CacheLifetimeResolver.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\HttpCache;
+
+use Cron\CronExpression;
+use Sulu\Component\Content\Metadata\CacheLifetimeMetadata;
+
+/**
+ * The cache lifetime resolver resolves the given cache lifetime metadata based on the type
+ * and returns an absolute cache lifetime in seconds.
+ */
+class CacheLifetimeResolver implements CacheLifetimeResolverInterface
```

Comment:
also the type check could go in here. then this class would be extensible without changing the metadata class.